### DISCUSSION TITLE
Do we have any java library to work with git lfs

### DISCUSSION THREAD

**Discussion by mantri-mounika** [git-lfs/git-lfs#4892] [Q&A]
Hi Team,

We are exploring java client which to connect to git server and perform git lfs operations like acquire the lock and release the lock.
We see the REST APIs instead we are looking for java library. Could you please help us.

**Comment by bk2204** [comment]
Hey,

I'm not aware of any Java library, but there are [API docs](https://github.com/git-lfs/git-lfs/tree/main/docs/api) if you'd like to write your own.  The API isn't very difficult to use, so I imagine it shouldn't be hard to write a small wrapper library that meets your needs.

**Reply by mantri-mounika** [reply]
Hi ,

Thanks for response.
I have a requirement to install using java on repository. Is there any API for that requirement ?

**Reply by bk2204** [reply]
No, we don't provide a Java API to install Git LFS.  Installation of software is best handled by the system package manager.

If you're looking for a Java API to manage Git LFS in any way, then I'm not aware of one.  Again, you could write your own small client in Java without much difficulty.

**Reply by mantri-mounika** [reply]
In the TortoiseSVN there is a concept called local lock and repository lock. First we check local lock , if  local lock is not present then we go for repository lock for better performance incase if we want to call this API for multiple times.
Does Git LFS Locking support local and repository locks ?

**Reply by bk2204** [reply]
No, there are only remote locks.  All lock operations have to go to the server.

**Comment by mantri-mounika** [comment]
Thank you for your Response

**Reply by mantri-mounika** [reply]
![image](https://user-images.githubusercontent.com/33999572/170020716-c571fca6-d203-4c1e-9f13-8b575c4cf810.png)
I am using POST request to create lock on a file but in the response I don't see any owner name who has locked that file. But if I see schema for createLock there is owner in response.
Is any param I am missing for getting owner in POST response? Could you please help me

**Reply by bk2204** [reply]
As the documentation says, the `owner` field is optional.  If you need it and it's not available, you'd want to list the locks using that API and find the information there.

**Reply by mantri-mounika** [reply]
Hi
While calling REST API to create lock , can we pass any extra information like comment or custom attribute with some value?
If yes could you please suggest sample schema or if no , any other alternative?

**Reply by bk2204** [reply]
The API doesn't provide a way to add a comment.  The Git LFS client doesn't provide custom attributes of any sort.  You'd want to examine alternatives for storing this or come up with a proposal where users could optionally add a comment if they wanted (which we would likely accept).

**Reply by mantri-mounika** [reply]
If this feature of adding a comment is required then who can provide this feature?
Do we need to create any feature request?

**Reply by bk2204** [reply]
You'd want to follow the contributing documentation for suggesting a new feature.  Once it's been discussed and decided that it's a good idea, we'd accept a patch for doing so or possibly implement it ourselves.

**Comment by slonopotamus** [comment]
I am a bit late to the party, but there is such library: https://github.com/git-as-svn/git-lfs-java
